tiny-event-sourcing-spring-app/src/main/kotlin/ru.quipy/demo/UserAggregate.kt

```diff
@@ -0,0 +1,98 @@
+package ru.quipy.demo
+
+import ru.quipy.core.AggregateType
+import ru.quipy.domain.Aggregate
+import java.util.*
+
+@AggregateType(aggregateEventsTableName = "aggregate-user")
+data class UserAggregate(
+    override val aggregateId: String
+) : Aggregate {
+    override var createdAt: Long = System.currentTimeMillis()
+    override var updatedAt: Long = System.currentTimeMillis()
+
+    var userName: String = ""
+    var userLogin: String = ""
+    var userPassword: String = ""
+    lateinit var defaultPaymentId: UUID
+    lateinit var defaultAddressId: UUID
+    var paymentMethods = mutableMapOf<UUID, PaymentMethod>()
+    var deliveryAddresses = mutableMapOf<UUID, DeliveryAddress>()
+}
+
+data class DeliveryAddress(
+    val addressId: UUID,
+    val address: String
+)
+
+data class PaymentMethod(
+    val paymentMethodId: UUID,
+    val cardNumber: String
+)
+
+fun UserAggregate.createUserCommand(
+    name: String,
+    password: String,
+    login: String
+): UserCreatedEvent {
+
+    return UserCreatedEvent(
```

Comment:
lets add validations: all fields not empty, some of them having more than N letters?

Reply:
I've added following code to command: if (name.length<3) {
        throw IllegalArgumentException("name is too small")
    }
    if (login.length<3) {
        throw IllegalArgumentException("login is too small")
    }
    if (password.isBlank()) {
        throw IllegalArgumentException("Can't change password: empty provided")
    }
    if (password.length < 8) {
        throw IllegalArgumentException("Password is too weak")
    }

Reply:
Add Id's to exceptions